import{_ as d,r as i,o as s,c as n,a as e,b as t,e as o,d as r}from"./app.0d17f42f.js";const c={},l=e("h1",{id:"import-and-export",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#import-and-export","aria-hidden":"true"},"#"),t(" Import and export")],-1),h=e("p",null,[e("em",null,"Many projects start with imports, and target exports. Here is where you find what is possible in TaxonWorks.")],-1),p=e("h2",{id:"imports",tabindex:"-1"},[e("a",{class:"header-anchor",href:"#imports","aria-hidden":"true"},"#"),t(" Imports")],-1),m={href:"https://speciesfilegroup.org/events.html",target:"_blank",rel:"noopener noreferrer"},u=r('<h3 id="data-types" tabindex="-1"><a class="header-anchor" href="#data-types" aria-hidden="true">#</a> Data types</h3><p>For bulk/batch uploads, TW supports import of various types of data including checklists, occurrence specimen record data, sequence data, source information (i.e. publications) and more. And using the DwC Archive Import Task, TW offers the use of the current gold standard for sharing specimen occurrence data via a tab-separated text file. This file can stand alone or be part of a Darwin Core Archive (DwCA) zipped file. These (your) data are mapped to Darwin Core terms, TaxonWorks specific elements, or your own custom Data Attributes (details below). Support for importing observations is underway.</p><ul><li>sample checklist (link)</li><li>sample specimen record dataset (link)</li></ul><h3 id="batch-imports" tabindex="-1"><a class="header-anchor" href="#batch-imports" aria-hidden="true">#</a> Batch imports</h3><p>There are various batch importers available within the UI (user interface). These are polished to differing degrees and have various benefits and limitations. The required format is described in the UI. All batch loaders are two-step, allowing for (and requiring) a preview of results before inserting them into the database.</p><ul><li>To explore available batch loaders click on a Data card in the Hub. If batch loader(s) are available then then the batch load link will be enabled.</li><li>Batch importers largely target tab-separated text files, though this is not exclusively the case.</li><li>Notable batch loaders are found in the TaxonNames, Otus, and Sources data cards, though others exist.</li><li>Explore various batch loaders (each data card highlighted in yellow has associated batch loaders at this writing).</li></ul><h4 id="try-batch" tabindex="-1"><a class="header-anchor" href="#try-batch" aria-hidden="true">#</a> Try batch</h4><p>In your test project,</p><ol><li>Go to the data tab</li><li>Select the Otu Data card</li><li>Click “batch load”</li><li>See instructions in the UI for expected / accepted data types and format.</li><li>Create your own file or use this <a href="link">test file</a> Header column = otu_name Blank lines are skipped Tab-delimited format, UTF-8 encoding, Unix line-endings required</li><li>Browse to your file to select it, click preview</li><li>If data looks as expected, browse to select that file again and click create.</li></ol><p>Batch loaders (as of March 2022) include:</p>',10),f={href:"https://docs.taxonworks.org/about/glossary.html#otu-operational-taxonomic-unit",target:"_blank",rel:"noopener noreferrer"},b=e("ul",null,[e("li",null,"simple batchload"),e("li",null,"data attributes"),e("li",null,"simple batch file"),e("li",null,"OTU with identifier batch load")],-1),g=r("<li>collecting events <ul><li>gpx (collecting events with georeferences)</li><li>castor</li></ul></li><li>collection objects <ul><li>castor</li><li>buffered strings</li></ul></li><li>descriptors <ul><li>qualitative descriptors</li><li>modify gene descriptor</li></ul></li><li>sequences <ul><li>Genbank</li><li>Genbank batch</li><li>primers</li></ul></li><li>sources <ul><li>BibTeX</li></ul></li><li>taxon names <ul><li>simple</li><li>castor</li></ul></li><li>asserted distributions <ul><li>simple</li></ul></li><li>namespaces <ul><li>simple</li></ul></li><li>sequence relationships <ul><li>primers batch</li></ul></li>",9),y=r('<h3 id="record-by-record" tabindex="-1"><a class="header-anchor" href="#record-by-record" aria-hidden="true">#</a> Record by record</h3><p>When first learning TaxonWorks, entering records one-at-a-time offers you the opportunity to learn about more of the features in TW and get a feel for how you and others experience the UI.</p><p>For example, you want to enter a specimen record. You have two Tasks enabling you to do this. Choose to use Comprehensive Specimen Digitization Task or the Simple New Specimen Task.</p><h4 id="try-simple-new-specimen" tabindex="-1"><a class="header-anchor" href="#try-simple-new-specimen" aria-hidden="true">#</a> Try Simple New Specimen</h4><p>In your project, try creating a simple new specimen record.</p><ul><li><p>Note you will need to select a namespace. You may find you need to add a namespace before you can do this TW task. Adding a value for namespace ensures your uploaded data records will be unique inside your TW project and across TW projects. In your project, you may also need more than one namespace. [Use Tommy’s INHS Insect Collection as an example, with 12 different namespaces that effectively group the various collections housed at INHS ENT].</p></li><li><p>If you tried the OTU batch loader you can pick one of your OTUs for the name to assign to this specimen.</p></li><li><p>Add an image if you wish</p></li><li><p>Select the Preparation type for this specimen. You may need to add a new value to the dropdown using the New preparation type task.</p></li></ul><h3 id="darwin-core-archive-dwc-a-import" tabindex="-1"><a class="header-anchor" href="#darwin-core-archive-dwc-a-import" aria-hidden="true">#</a> Darwin Core Archive (DwC-A) import</h3><p>To upload occurrence (vouchered specimen data records) data, TW offers the ability to use a DwC Archive file. This provides options to share data that maps to concepts (aka terms, elements, properties) in Darwin Core, but also to specific fields in TaxonWorks and to custom &quot;Data Attribute&quot; fields you can add, if needed.</p><p>For these <strong>occurrence datasets</strong> one needs to map each term to be imported. This next section explains the different mapping options available and steps needed to get started.</p><h3 id="precursor-steps-to-dwc-a-import" tabindex="-1"><a class="header-anchor" href="#precursor-steps-to-dwc-a-import" aria-hidden="true">#</a> Precursor steps to DwC-A import</h3><ul><li><p>For your dataset, you may need to create one or more <strong>namespaces.</strong> These namespaces allow TW to</p><ul><li>track uniqueness of each object and</li><li>group these objects.</li></ul><p>For example, in your collection of physical objects, your catalog numbers may have duplicates if your specimens have come from other instituion collections in the past. Creating namespaces in TW is simple to do, but spending some time on this before upload can simplify the process. Namespaces also need to be unique in your TW Project (first-come, first-served). [NEED EXAMPLES here].</p></li><li><p>In addition, if your dataset has terms not mappable to TW concepts or the DwC terms we currently support, you will need to create custom &quot;Data Attributes.&quot; These need to be created before upload (ideally). If you don&#39;t, then records with these data will not upload but are easy for you to address after other records import.</p></li><li><p>For some <strong>biocuration</strong> data in TW you may find you need to add the group and accepted values for a given class before upload (e.g. group=caste and class=queen).</p></li></ul><h3 id="dwc-terms-mappings" tabindex="-1"><a class="header-anchor" href="#dwc-terms-mappings" aria-hidden="true">#</a> DwC terms mappings</h3><p>Next please note the dwc terms that TW currently recognizes. Also see examples below for how to map to existing TW terms, custom data attributes, and biocuration information.</p><h4 id="record-level-class" tabindex="-1"><a class="header-anchor" href="#record-level-class" aria-hidden="true">#</a> Record-level class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>type</code></td><td>It is checked that it equals <code>PhysicalObject</code> before allowing the record to be imported. If the value is empty or term not present it is assumed it is a <code>PhysicalObject</code></td></tr><tr><td><code>institutionCode</code></td><td>Selects the repository for the specimen that is registered with an acronym equal to this value</td></tr><tr><td><code>collectionCode</code></td><td>Paired with <code>institutionCode</code> it is used to select the namespace for <code>catalogNumber</code> from a user-defined lookup table in import settings, the value itself is not imported.</td></tr><tr><td><code>basisOfRecord</code></td><td>It is checked that it equals an expected valid value for term, e.g. <code>PreservedSpecimen</code> or <code>FossilSpecimen</code> before allowing the record to be imported. If the value is empty or term not present it is assumed it is a <code>PreservedSpecimen</code> [IS THIS STILL TRUE?]</td></tr></tbody></table><h4 id="occurrence-class" tabindex="-1"><a class="header-anchor" href="#occurrence-class" aria-hidden="true">#</a> Occurrence class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>catalogNumber</code></td><td>The identifier value for Catalog Number local identifier. The namespace is selected from the namespaces lookup table in import settings queried by <code>institutionCode</code>:<code>collectionCode</code> pair.</td></tr><tr><td><code>recordedBy</code></td><td>It is imported as-is in verbatim collectors field of the collecting event. Additionally, the value is parsed into people and assigned as collectors of the CE. Previously existing people are not used unless the data origin is the same dataset the record belongs to, otherwise any missing people are created.</td></tr><tr><td><code>individualCount</code></td><td>The total number of entities associated with the specimen record (e.g. this record may be for a &quot;lot&quot; containing 6 objects).</td></tr><tr><td><code>sex</code></td><td>Selects the biocuration class from the &quot;sex&quot; biocuration group to be assigned as biocuration classification for the specimen.</td></tr><tr><td><code>preparations</code></td><td>Selects an existing preparation matching the name with this value.</td></tr></tbody></table><h4 id="event-class" tabindex="-1"><a class="header-anchor" href="#event-class" aria-hidden="true">#</a> Event class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>fieldNumber</code></td><td>Verbatim trip identifier of collecting event</td></tr><tr><td><code>eventDate</code></td><td>The ISO8601-formatted date is split into start year, month and day collecting event fields. If the value is composed of two dates separated by <code>/</code>, then rightmost date is used as end date and split in the same way as start date. If data contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>eventTime</code></td><td>Time is split into time start hour, minute, and second of collecting event</td></tr><tr><td><code>startDayOfYear</code></td><td>Using <code>year</code> and the value for this term month and day are calculated and stored in start year, month, and day collecting event fields. If the computed value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>endDayOfYear</code></td><td>Using <code>year</code> and the value for this term month and day are calculated and stored in end year, month and day collecting event fields. If the computed value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>year</code></td><td>The start date year of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>month</code></td><td>The start date month of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import.</td></tr><tr><td><code>day</code></td><td>The start date day of the collecting event. If the value contradicts dates from other non-empty date-related terms the record will fail to import</td></tr><tr><td><code>verbatimDate</code></td><td>Verbatim date of the collecting event</td></tr><tr><td><code>habitat</code></td><td>Verbatim habitat of the collecting event</td></tr><tr><td><code>samplingProtocol</code></td><td>Verbatim method of the collecting event</td></tr><tr><td><code>fieldNotes</code></td><td>Field notes of the collecting event</td></tr></tbody></table><h4 id="location-class" tabindex="-1"><a class="header-anchor" href="#location-class" aria-hidden="true">#</a> Location class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>fieldNumber</code></td><td>Verbatim trip identifier of collecting event</td></tr></tbody></table><h4 id="identifcation-class" tabindex="-1"><a class="header-anchor" href="#identifcation-class" aria-hidden="true">#</a> Identifcation class</h4>',22),v=e("thead",null,[e("tr",null,[e("th",null,"Term"),e("th",null,"Mapping")])],-1),w=e("tr",null,[e("td",null,[e("code",null,"identifiedBy")]),e("td",null,[t("A list (concatenated and separated) of names of people, groups, or organizations who assigned the Taxon to the subject. If possible, separate the values in a list with space vertical bar space | (known as a pipe). (e.g. "),e("code",null,"Theodore Pappenfuss | Robert Macey"),t(")")])],-1),_=e("td",null,[e("code",null,"dateIdentified")],-1),x={href:"https://dwc.tdwg.org/terms/#dwc:dateIdentified",target:"_blank",rel:"noopener noreferrer"},T=r('<h4 id="taxon-class" tabindex="-1"><a class="header-anchor" href="#taxon-class" aria-hidden="true">#</a> Taxon class</h4><table><thead><tr><th>Term</th><th>Mapping</th></tr></thead><tbody><tr><td><code>nomenclaturalCode</code></td><td>Selects the nomenclatural code for the taxon ranks used when creating protonyms. The value itself is not imported</td></tr><tr><td><code>kingdom</code></td><td>Creates (unless already present) a protonym at kingdom rank</td></tr><tr><td><code>phylum</code></td><td>Creates (unless already present) a protonym at phylum rank</td></tr><tr><td><code>class</code></td><td>Creates (unless already present) a protonym at class rank</td></tr><tr><td><code>order</code></td><td>Creates (unless already present) a protonym at order rank</td></tr><tr><td><code>family</code></td><td>Creates (unless already present) a protonym at family rank</td></tr><tr><td><code>genus</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>subgenus</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>specificEpithet</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>infraspecificEpithet</code></td><td>Ignored. Extracted from <code>scientificName</code> instead</td></tr><tr><td><code>scientificName</code></td><td>Several protonyms created (only when not present already) with their corresponding ranks and placements</td></tr><tr><td><code>taxonRank</code></td><td>The taxon rank of the most specific protonym</td></tr><tr><td><code>higherClassification</code></td><td>Several protonyms created (only when not present already) with their corresponding ranks and placement. In case a protonym was not already present, only family-group names will be created, names with classsification higher than family-group not previously registered will result in error. Names at genus rank or lower are ignored and extracted from <code>scientificName</code> instead</td></tr><tr><td><code>scientificNameAuthorship</code></td><td>Verbatim author of most specific protonym</td></tr></tbody></table><h3 id="tw-non-standard-mappings" tabindex="-1"><a class="header-anchor" href="#tw-non-standard-mappings" aria-hidden="true">#</a> TW non-standard mappings</h3><p>The DwC importer task includes some TW-specific mappings that are neither DwC core terms nor in any DwC extension term lists but instead, direct mappings to predicates in your projects imported as data attributes for collection objects and collecting events, biocuration groups and classes, and as an advanced-use feature you may have direct mappings to model fields.</p><p><strong>IMPORTANT</strong>: If submitting an actual DwC-A zip file and not tab-separated text file or spreadsheet, this TW-specific mappings have to be placed as headers in the core table, and not in meta.xml. If you are replacing a mapping from meta.xml, you must make sure to comment it out and also if inserting colums make sure you do the appropriate adjustments to avoid collision.</p><h4 id="mappings-to-project-predicates" tabindex="-1"><a class="header-anchor" href="#mappings-to-project-predicates" aria-hidden="true">#</a> Mappings to project predicates</h4><p>In cases where you need to import predicate values targetting the imported collection object or collecting event you may do so by naming the column with a pattern like <code>TW:DataAttribute:&lt;target_class&gt;:&lt;predicate_identifier&gt;</code>. <code>&lt;target_class&gt;</code> may be <code>CollectionObject</code> or <code>CollectingEvent</code>, and the <code>&lt;predicate_identifier&gt;</code> may be the either the name of the predicate or its URI. As an example if you have a predicate registered with name <code>ageInDays</code> and URI <code>http://rs.gbif.org/terms/1.0/ageInDays</code>, both <code>TW:DataAttribute:CollectionObject:ageInDays</code> and <code>TW:DataAttribute:CollectionObject:http://rs.gbif.org/terms/1.0/ageInDays</code> can be used to refer to the same predicate.</p><h4 id="mappings-to-biocuration-groups-and-classes" tabindex="-1"><a class="header-anchor" href="#mappings-to-biocuration-groups-and-classes" aria-hidden="true">#</a> Mappings to biocuration groups and classes</h4><p>The importer is able to map <code>sex</code> into the appropriate biocuration group and select the approriate class according to the value. For additional mappings you may use a special column name pattern to select a biocuration group like <code>TW::BiocurationGroup:&lt;group_identifier&gt;</code> where <code>&lt;group_identifier&gt;</code> can be the name of the biocuration group or its URI. In addition the values must match an existing biocuration class and you may use either its name or URI. For example, if you have a biocuration group registered with name <code>Caste</code> and URI <code>urn:example:ants:caste</code> and biocuration class with name <code>Queen</code> and URI <code>urn:example:ants:caste:queen</code> the following examples do all create the same biocuration classification:</p><table><thead><tr><th>Caste</th><th>urn:example:ants:caste</th></tr></thead><tbody><tr><td>Queen</td><td>urn:example:ants:caste:queen</td></tr><tr><td>urn:example:ants:caste:queen</td><td>Queen</td></tr></tbody></table><h4 id="mappings-to-dwc-predicates" tabindex="-1"><a class="header-anchor" href="#mappings-to-dwc-predicates" aria-hidden="true">#</a> Mappings to DwC predicates</h4><p>Whenever the importer sees that your project has custom attributes for collecting events and/or collection objects that matches Darwin Core URI terms (<code>http://rs.tdwg.org/dwc/terms/&lt;term&gt;</code>), them will be imported as data attributes regardless of any existing mapping of the same field. This allows to preserve verbatim dataaset value for reference as also to import data from terms not supported by the importer.</p><h4 id="direct-mapping-to-tw-model-fields-advanced" tabindex="-1"><a class="header-anchor" href="#direct-mapping-to-tw-model-fields-advanced" aria-hidden="true">#</a> Direct mapping to TW model fields (advanced)</h4>',13),k=e("code",null,"TW:<model_class>:<field>",-1),I={href:"https://docs.taxonworks.org/develop/Data/models.html#collection-object",target:"_blank",rel:"noopener noreferrer"},C=e("code",null,"CollectionObject",-1),D={href:"https://docs.taxonworks.org/develop/Data/models.html#collecting-event",target:"_blank",rel:"noopener noreferrer"},W=e("code",null,"CollectingEvent",-1),j=e("code",null,"<field>",-1),N=r('<table><thead><tr><th>Class</th><th>fields</th></tr></thead><tbody><tr><td><code>CollectionObject</code></td><td><code>buffered_collecting_event</code>, <code>buffered_determinations</code>, <code>buffered_other_labels</code>, <code>total</code>,</td></tr><tr><td><code>CollectingEvent</code></td><td><code>document_label</code>, <code>print_label</code>, <code>verbatim_label</code>, <code>field_notes</code>, <code>formation</code>, <code>group</code>, <code>lithology</code>, <code>max_ma</code>, <code>maximum_elevation</code>, <code>member</code>, <code>min_ma</code>, <code>minimum_elevation</code>, <code>elevation_precision</code>, <code>start_date_day</code>, <code>start_date_month</code>, <code>start_date_year</code>, <code>end_date_day</code>, <code>end_date_month</code>, <code>end_date_year</code>, <code>time_end_hour</code>, <code>time_end_minute</code>, <code>time_end_second</code>, <code>time_start_hour</code>, <code>time_start_minute</code>, <code>time_start_second</code>, <code>verbatim_collectors</code>, <code>verbatim_date</code>, <code>verbatim_datum</code>, <code>verbatim_elevation</code>, <code>verbatim_geolocation_uncertainty</code>, <code>verbatim_habitat</code>, <code>verbatim_latitude</code>, <code>verbatim_locality</code>, <code>verbatim_longitude</code>, <code>verbatim_method</code>, <code>verbatim_trip_identifier</code></td></tr></tbody></table><h2 id="exports" tabindex="-1"><a class="header-anchor" href="#exports" aria-hidden="true">#</a> Exports</h2><h2 id="requesting-a-new-import-or-export-type" tabindex="-1"><a class="header-anchor" href="#requesting-a-new-import-or-export-type" aria-hidden="true">#</a> Requesting a new import or export type</h2>',3);function q(S,A){const a=i("ExternalLinkIcon");return s(),n("div",null,[l,h,p,e("p",null,[t("TaxonWorks offers two ways to import data: 1) via the User Interface (UI) entering one record at a time or batch uploading, and 2) programmatically with customized scripts for very large datasets (> 1000 records). We seek to make the data import and entry as savvy as possible. As you experience these options, please contribute your ideas for ways to enhance these processes. (Submit ideas, submit code, and please bring your voice to our "),e("a",m,[t("sfgcommons meetings on Wednesdays"),o(a)]),t(").")]),u,e("ul",null,[e("li",null,[t("OTUs "),e("a",f,[t("operational taxonomic units"),o(a)]),b]),g]),y,e("table",null,[v,e("tbody",null,[w,e("tr",null,[_,e("td",null,[t("The date on which the subject was determined as representing the Taxon. Best practice is to use a date that conforms to ISO 8601-1:2019 "),e("a",x,[t("see examples"),o(a)]),t(".")])])])]),T,e("p",null,[t("This is an advance mapping and requires knowledge of the underlying TW models. The pattern is "),k,t(" where model can be either "),e("a",I,[C,o(a)]),t(" or "),e("a",D,[W,o(a)]),t(", and "),j,t(" can be the ones listed below.")]),N])}const U=d(c,[["render",q],["__file","import-and-export.html.vue"]]);export{U as default};
